### ISSUE TITLE
compare two word documents and generating the changes not working

### ISSUE THREAD

**Issue by manikantad** [OfficeDev/Open-Xml-PowerTools#108]
while comparing document facing issue .
 i got 2 issues for different files
for document 1.file unsupported :document contains text boxes. 
for document 2.: file contains correpted data.

can you please help me to get a solution for this.
![img-0 2](https://cloud.githubusercontent.com/assets/22674966/22096082/92cf1bf4-de3f-11e6-8a1f-9710100ef20a.png)

**Comment by iivel** [comment]
I ran into this on a project I'm converting from Interop as well. Shut that down in its tracks.  You'll note that it's specified as unsupported "for now" in the code.

        // prohibit
        // - altChunk
        // - subDoc
        // - contentPart
        // - text boxes and nested tables (for now)
        private static void TestForInvalidContent(WordprocessingDocument wDoc)

Any word on getting text boxes and tables fixed?

**Comment by EricWhiteDev** [comment]
@manikantad @iivel 

Hi Guys, fixing this is (most probably) in the near future.  The algorithm that the module takes is one I devised after seeing the intrinsic problems with the stock LCS algorithm for content that contains both paragraphs and tables/rows/cells.  (Embedded text boxes is a degenerate, simpler case of block-level content that can contain block-level content.)  Having this module be perfect is a high priority for me.

I made a mistake when devising that algorithm, such that it wasn't possible to 'reconstitute' the original document from the internal flat referential model if the document contained nested block-level content controls.   (It was possible sometimes, but there are edge cases where it doesn't work.)  I ran out of time on that project, and further, it was not necessary or required for that project, so the project was stopped at that point.  I need to re-work the algorithm one more time (this will be #4).

My schedule will firm up soon.  I am hopeful that this will be next on my schedule, but there are no guarantees.  :-)

Best, Eric

**Comment by iivel** [comment]
@EricWhiteDev 

Thanks Eric. This is an awesome project and I certainly appreciate your efforts on it.  FWIW, my approach is based on trying to replicate the output of the interop-based application I need to replace.

http://levii.com/47/merging-more-than-2-ms-word-documents-what-a-pain

As far as the algorithm I was looking at (after the WmlComparer does it's thing) I do have a nearly exact match from the commit I mentioned earlier. I forced a return from the method instead of throwing an exception && that gets me close enough to move forward.  I'm just stuck on tiny part and was wondering if you had a quick minute for a suggestion.  I've attached the modified example class for the comparer and copious notes of where I'm stuck (the right way to handle the foreach at line 81). 

Thanks again either way, this project is a wonderful resource!

[WmlComparer01.txt](https://github.com/OfficeDev/Open-Xml-PowerTools/files/736986/WmlComparer01.txt)
